test/integration/targets/docker_swarm_facts/tasks/test_swarm_facts.yml

```diff
@@ -0,0 +1,41 @@
+---
+- block:
+  - name: Make sure we're not already using Docker swarm
+    docker_swarm:
+      state: absent
+      force: true
+
+  - name: Try to get docker_swarm_facts when docker is not running in swarm mode
+    docker_swarm_facts:
+    ignore_errors: yes
+    register: output
+
+  - name: assert failure when called when swarm is not in use or not run on mamager node
+    assert:
+      that:
+         - 'output.failed'
+         - 'output.msg == "Error running docker swarm module: must run on swarm manager node"'
+
+  - name: Create a Swarm cluster
+    docker_swarm:
+      state: present
+    register: output
+
+  - name: assert changed when create a new swarm cluster
+    assert:
+      that:
+         - 'output.changed'
+         - 'output.actions[0] | regex_search("New Swarm cluster created: ")'
+         - 'output.swarm_facts.JoinTokens.Manager'
+         - 'output.swarm_facts.JoinTokens.Worker'
+
+  - name: Try to get docker_swarm_facts when docker is running in swarm mode and as manager
+    docker_swarm_facts:
+    ignore_errors: yes
+    register: output
```

Comment:
You should investigate the output (and probably remove `ignore_errors: yes`).

Some things to check:
```
         - output.can_talk_to_docker == true
         - output.docker_swarm_active == true
         - output.docker_swarm_manager == true
```

Reply:
Sure that code is before the #52189, so this is cosmetic if generally it is what it is supposed to be :)